src/audio/dai-zephyr.c

```diff
@@ -1224,12 +1224,26 @@ static int dai_copy(struct comp_dev *dev)
 
 	comp_dbg(dev, "dai_copy()");
 
-	/* get data sizes from DMA */
+	/* DMA status can return -EPIPE and current status if xrun occurs, then it is not critical
+	 * and flow shall continue. Other error values will be treated as critical.
+	 */
 	ret = dma_get_status(dd->chan->dma->z_dev, dd->chan->index, &stat);
-	if (ret < 0) {
+	switch (ret) {
+	case 0:
+		break;
+	case -EPIPE:
+		if (dev->direction == SOF_IPC_STREAM_PLAYBACK)
+			comp_warn(dev, "dai_copy(): dma_get_status() underrun occurred, ret = %u",
+				  ret);
+		else
+			comp_warn(dev, "dai_copy(): dma_get_status() overrun occurred, ret = %u",
+				  ret);
+		break;
+	default:
 		dai_report_xrun(dev, 0);
```

Comment:
@kv2019i IIRC calling ```dai_report_xrun(dev, 0);``` would cause the XRUN handler on SOF Linux driver to run (and stop/restart the stream) ? Should this not be something to enable/disable (default) at runtime ?